### ISSUE TITLE
Authentication on websocket

### ISSUE THREAD

**Issue by lintardec** [cypress-io/cypress#2335]
### Bug

No authentication happening when calling an authenticated websocket.

### Current behavior:

My application uses an authenticated websocket connection. I have it authenticating using BASIC in dev which the browser does for me.

When connecting cypress I get 401 unauthorised responses

### Desired behavior:

For the websocket connection via cypress to perform the same authentication as the cy.visit request does

### Steps to reproduce:

My test is currently just trying to connect:
```
describe('My First Test', function() {
  it('Visits the Kitchen Sink', function() {

    cy.visit('http://localhost:8081/', {
      auth: {
        username: 'admin',
        password: 'password'
      },
    })
  })
})
```

which works.

But then all the content of the page is governed by what the websocket comes back with. My app trys to open a websocket:
      socket = new WebSocket('ws://localhost:8081/socket')

But I just get this error:

WebSocket connection to 'ws://localhost:8081/socket' failed: HTTP Authentication failed; no valid credentials available

It needs to pass through a header like: 
    Authorization: Basic c3VwZXI6c3VwZXI=

Have also tried under HTTPS / WSS with the same results

### Versions
Cypress: 3.0.3
operating system: windows 7
browser: chrome 68

**Comment by paulbellamy** [comment]
As far as I can tell this is not actually a duplicate of #2492. That issue is about being able to mock out websocket connections, whereas this is about correctly passing through the Authorization header on real websocket requests. Currently the basic auth header is set in the network proxy, but websocket requests don't go through the network proxy, so they never get the basic auth headers set.

I'm new to using cypress, so apologies if that's incorrect.

**Comment by lintardec** [comment]
That is correct. I was trying to make an actual websocket connection with an authentication header.

I have now mocked the socket connection to overcome the issue, not ideal though!

**Comment by jgarcia1095** [comment]
@jennifer-shehane , I have the same problem "WebSocket connection to 'wss://????????????????????' failed: HTTP Authentication failed; no valid credentials available", when I am doing the test, my front application generated an error when going through the web socket, however the error happens without using any mock on the web socket, to complement the above, I do not have the problem if I use version 5.0 but in later versions such as 6.5 is presented, using exactly the same code
